sklearn/tests/test_pipeline.py

```diff
@@ -983,6 +983,50 @@ def test_set_feature_union_step_drop():
     assert not record
 
 
+def test_set_feature_union_passthrough():
+    mult2 = Mult(2)
+    mult2.get_feature_names = lambda: ["x2"]
+    mult3 = Mult(3)
+    mult3.get_feature_names = lambda: ["x3"]
+    X = np.asarray([[1]])
+
+    ft = FeatureUnion([("m2", mult2), ("m3", mult3)])
+    assert_array_equal([[2, 3]], ft.fit(X).transform(X))
+    assert_array_equal([[2, 3]], ft.fit_transform(X))
+    assert ["m2__x2", "m3__x3"] == ft.get_feature_names()
+
+    with pytest.warns(None) as record:
+        ft.set_params(m2="passthrough")
+        assert_array_equal([[1, 3]], ft.fit(X).transform(X))
+        assert_array_equal([[1, 3]], ft.fit_transform(X))
+    assert ["m3__x3"] == ft.get_feature_names()
```

Comment:
Since, passthrough does not have `get_feature_names`, I have removed the feature names tests from this. I initially did those because they were done fot `drop` but there it makes sense. I think checking the feature names is not needed here.